runtime/POSIX/fd.c

```diff
@@ -128,47 +129,38 @@ static int has_permission(int flags, struct stat64 *s) {
   gid_t egid = getegid();
 #endif  
 
-  if (read_access && ((mode & S_IRUSR) | (mode & S_IRGRP) | (mode & S_IROTH)))
+  if (read_access && !((mode & S_IRUSR) | (mode & S_IRGRP) | (mode & S_IROTH))) {
```

Comment:
Interesting, why was `!` missing? KLEE bug??

Reply:
I think so.

Reply:
Probably should make a note of that and file it. When I have time...